packages/models/src/store/merge-strategy.ts

```diff
@@ -0,0 +1,266 @@
+      ],
+    }
+  }
+}
+
+export class DefaultMergeStrategy<S> extends MergeStrategy<S, S> {
+  private defaultMergeWith = mergeWith((objValue, srcValue) => {
+    if (Array.isArray(objValue)) {
+      return objValue.concat(srcValue)
+    }
+  })
+
+  merge(object: S, source?: S): S {
+    if (source === undefined || source === null) return object
+    if (Array.isArray(object)) return object.concat(source) as S
+    return this.defaultMergeWith(source, object)
+  }
+
+  private findAndUpdateInState(
+    paths: string[],
+    value: unknown,
+    state: S,
+  ): {
+    removePath?: string[]
+    state?: S
+  } {
+    if (typeof state !== 'object' || state === null || !paths.length)
+      return { state: value as S, removePath: Array.isArray(value) ? paths : undefined }
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let current: any = state
+    for (let idx = 0; idx < paths.length; idx++) {
+      if (paths[idx] in current) {
+        current = current[paths[idx]]
+        if (Array.isArray(current)) {
+          if (idx === paths.length - 1) {
+            // need to remove others array, or otherwise they will merge into new mergedState
+            return {
+              removePath: paths,
+              state: set(paths, value)(state),
+            }
+          } else {
+            // replace current state
+            return {
+              state: set(paths, value)(state),
+            }
+          }
+        }
+      } else {
+        return {}
+      }
+    }
+    return {
+      state: set(paths, value)(state),
+    }
+  }
+
+  private removeNotMergedState({
+    state,
+    isValueEqual,
+    isSimpleType,
+    changedPaths,
+    mergedState,
+  }: {
+    state: S
+    changedPaths: string[]
+    isValueEqual(value: unknown, compare: unknown): boolean
+    isSimpleType(value: unknown): boolean
+    mergedState?: S
+  }) {
+    deepForIn(state, (pathValue, currentPaths) => {
+      const valueInMerged = get(currentPaths)(mergedState)
+      const isChangedPaths =
+        currentPaths.length === changedPaths.length && currentPaths.every((path, idx) => changedPaths[idx] === path)
+      if (isValueEqual(pathValue, valueInMerged) || isChangedPaths) {
+        return false
+      }
+      if (isSimpleType(pathValue)) {
+        // is simple type and is not in mergedState or changed by current, remove it
+        if (currentPaths.length === 1) {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          delete (state as any)[currentPaths[0]]
+        } else if (currentPaths.length > 1) {
+          const lastLevelPath = currentPaths.slice(0, -1)
+          const lastLevelPathvalue = get(lastLevelPath)(state)
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          delete lastLevelPathvalue[currentPaths.at(-1)!]
+        } else {
+          // ignore if root is simple type, no need to handle
+        }
+        return false
+      }
+      return true
```

Comment:
The `continueDeepFor` should be defined independently for testing and understanding

Reply:
It depends on the outer variable `mergedState`. So I will add test cases for the `removeNotMergedState` function.

Reply:
If we can pass `outer variables` into it as a parameter to make the logic independent and easy to test